### PR TITLE
Site Dashboard: Always display all sites

### PR DESCRIPTION
**Pull request by arthur791004**
Related to https://github.com/Automattic/dotcom-forge/issues/7010

## Proposed Changes

* Query all sites instead of primary and recent sites to avoid only a couple of sites visible
* Ensure the sites are shown by the sorting method as much as possible so the order of the sites won't be changed after the first render.

## Testing Instructions

* Clear the `calypso_store`
  image
* Go to /sites?flags=force-sympathy
  * If you cannot reproduce the issue, you can block the `/me/sites` endpoint 
* Make sure you won't just see a couple of sites on the first render
* Make sure the order of sites won't be changed (it still has a chance...😓)

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

### CODE REVIEW

**Review by DustyReagan — COMMENTED** [review]
I think we lose scrolling in this PR. I tested in Firefox and Chrome and could not scroll down my list of sites after it finished loading.

Nit: Also, after the first page load, on subsequent refreshes, we see the paging select components with the spinner. Ideally, they would be hidden when the spinner is visible.

Outside of the above, I think it's an improved user experience!

**Review by DustyReagan — APPROVED** [review]
Works great! I can scroll now! 😆

File: client/components/route/index.js [hunk 1]
```diff
@@ -22,6 +22,11 @@ export function RouteProvider( {
 		() => ( { currentSection, currentRoute, currentQuery } ),
 		[ currentSection, currentRoute, currentQuery ]
 	);
+
+	if ( ! currentRoute ) {
```
Comment (taipeicoder) [inline client/components/route/index.js:26]: Question: What's the purpose of this change in the PR? Asking because this seems to be a significant change so I'd like to understand more about its impact 🙂

**Review by taipeicoder — APPROVED** [review]
Works as intended, left a non-blocking question.

File: client/components/route/index.js (hunk 1 above)
Comment (arthur791004) [reply client/components/route/index.js:26]: We use `currentRoute` to determine something when rendering. However, it might be `undefined` (it's weird) on the first render and it leads to inconsistency.

For example, the following condition was false on the first render, so the component was loading the primary and recent sites on the first render. Later, the value became true, and the component changed to load all sites. Without this change, the dashboard might still show a couple of sites first and it looked weird.
* `shouldQueryAllSites: currentRoute && currentRoute !== '/jetpack/connect/authorize'`
